modules/subscriptions.php

```diff
@@ -567,6 +567,26 @@ function Jetpack_Subscriptions_Widget() {
 		$this->WP_Widget( 'blog_subscription', __( 'Blog Subscriptions (Jetpack)', 'jetpack' ), $widget_ops, $control_ops );
 	}
 
+	/*
+	 * Check for the option to show subscriber count.
+	 *
+	 * This function only serves the purpose of backwards compatibility for versions < 3.4
+	 *
+	 * @since 3.5
+	 * @return bool  Will return true only if a site still has the option to show subscriber count.
+	 */
+	function has_show_subscriber_count_option() {
+		$widget_options = get_option( 'widget_blog_subscription' );
+		if ( ! empty( $widget_options ) ) {
+			foreach ( $widget_options as $k => $option ) {
```

Comment:
I'm 99% sure since we're storing the widget options as <a href="https://github.com/Automattic/jetpack/blob/5fa5796df44b1f26a0b9129d7e2b4a0f522d9ba9/modules/subscriptions.php#L722">an array</a> and then updating it with core's WP_Widget::save_settings(), which takes an array as the param.   

I will add a check to be safe.